### PR TITLE
Login: restore ability to use safari saved password

### PR DESCRIPTION
**Pull request by nheagy**
Refs #7272

Safari saved passwords are functional again. Properly logs in, and handles errors with grace.

safari-saved-demo

To test:
- login via safari saved passwords
- the same, with an incorrect passwords
- with and without requiring 2fa

Needs review: @aerych

### CODE REVIEW

File: WordPress/Classes/ViewRelated/NUX/LoginEmailViewController.swift [hunk 1]
```diff
@@ -198,7 +198,7 @@ class LoginEmailViewController: LoginViewController, SigninKeyboardResponder {
         loginFields.safariStoredUsernameHash = username.hash
         loginFields.safariStoredPasswordHash = password.hash
 
-        loginWithUsernamePassword(immediately: false)
+        loginWithUsernamePassword(immediately: true)
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/NUX/LoginEmailViewController.swift]: 👍, weird that we weren't doing this for safari stored passwords previously when we were doing it for 1password.

File: WordPress/Classes/ViewRelated/NUX/LoginEmailViewController.swift [hunk 2]
```diff
@@ -272,6 +272,13 @@ class LoginEmailViewController: LoginViewController, SigninKeyboardResponder {
         })
     }
 
+    override func displayRemoteError(_ error: Error!) {
+        configureViewLoading(false)
+
+        self.errorToPresent = error
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/NUX/LoginEmailViewController.swift]: No need for `self`

File: WordPress/Classes/ViewRelated/NUX/SigninEditingState.swift [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import Foundation
 
 class SigninEditingState {
-    static var signinEditingStateActive = false
+    static var signinEditingStateActive = true
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/NUX/SigninEditingState.swift]: Let's do this in a separate PR.  A quick pass through the wpcom password flow has the keyboard jumping up and down between navigation and it doesn't look too good.  I think its going to need more attention than this one line change.

File: WordPress/Classes/ViewRelated/NUX/NUXAbstractViewController.swift [hunk 4]
```diff
@@ -62,20 +62,6 @@ class NUXAbstractViewController: UIViewController, LoginSegueHandler {
         return UIDevice.isPad() ? .all : .portrait
     }
 
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/NUXAbstractViewController.swift]: Nope, won't be a problem, as the old login flow doesn't use segues at all.

File: WordPress/Classes/ViewRelated/NUX/SigninEditingState.swift (hunk 3 above)
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/SigninEditingState.swift]: I can remove this. Just to be clear, this only affects the first text field seen: the email text field. After that, behaviour is exactly as it was previously as `signinEditingStateActive` is set to `false` or `true` as the user moves through the flow.

The implication for Safari saved passwords is that users aren't prompted for them until they tap on the text field. It's a little bit jarring.

File: WordPress/Classes/ViewRelated/NUX/NUXAbstractViewController.swift (hunk 4 above)
Comment (aerych) [inline WordPress/Classes/ViewRelated/NUX/NUXAbstractViewController.swift]: 🤦‍♂️

File: WordPress/Classes/ViewRelated/NUX/SigninEditingState.swift (hunk 3 above)
Comment (aerych) [inline WordPress/Classes/ViewRelated/NUX/SigninEditingState.swift]: I went back and retested in the simulator with regular animations, slow animation, and then on a device for good measure.  I think what I was seeing before, super jumpy semi-glitchy keyboard animations was just the simulator being a pita (sorry for the noise!).  However, something I'm still seeing is this: 

simulator screen shot jul 12 2017 1 12 48 pm

Notice how the keyboard is visible but the view has not adjusted its bottom constraint. So, I'll amend my earlier request to just needing to get the view properly sizing.

Comment (aerych) [inline WordPress/Classes/ViewRelated/NUX/SigninEditingState.swift]: I should clarify that happens when the email vc first appears when opened from the prologue.  I can reproduce it both in the simulator and on a device.

Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/SigninEditingState.swift]: Ok, I reverted this because the _proper_ fix got well beyond the scope of this PR. I've got it sitting in a local branch in case I get time to finish it.

**Comment by nheagy** [comment]
@aerych thanks! this should be ready for another look 🤞

**Comment by aerych** [comment]
:shipit: sir!
